### PR TITLE
doc: Bluetooth spring cleaning 📜🧹

### PR DESCRIPTION
**Pull request by jori-nordic**
Re-organizes Bluetooth docs to be more `A  E  S  T  H  E  T  I  C`.
- rename of sections
- more links (we have html at home)
- more table of contents

Documentation build here: https://builds.zephyrproject.io/zephyr/pr/70679/docs/connectivity/bluetooth/index.html

~Stacked on top of https://github.com/zephyrproject-rtos/zephyr/pull/70656 (ie don't review the first two commits).~

### CODE REVIEW

**Review by jhedberg — DISMISSED** [review]

File: doc/connectivity/bluetooth/bluetooth-ctlr-arch.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .. _bluetooth-ctlr-arch:
 
-Bluetooth Low Energy Controller
-###############################
+Low Energy Controller
```
Comment (theob-pro) [inline doc/connectivity/bluetooth/bluetooth-ctlr-arch.rst:3]:
```suggestion
LE Controller
```

File: doc/connectivity/bluetooth/bluetooth-dev.rst [hunk 2]
```diff
@@ -31,10 +35,15 @@ one or another setup to match your needs.
 
 There are 4 possible hardware setups to use with Zephyr and Bluetooth:
```
Comment (theob-pro) [inline doc/connectivity/bluetooth/bluetooth-dev.rst:36]:
```suggestion
There are 3 possible setups to use with Zephyr and Bluetooth:
```

File: doc/connectivity/bluetooth/bluetooth-le-host.rst [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+Persistent storage
+==================
+
+The Bluetooth host stack uses the settings subsystem to implement
+persistent storage to flash. This requires the presence of a flash
+driver and a designated "storage" partition on flash. A typical set of
+configuration options needed will look something like the following:
+
+  .. code-block:: none
```
Comment (theob-pro) [inline doc/connectivity/bluetooth/bluetooth-le-host.rst:193]:
```suggestion
  .. code-block:: cfg
```

File: doc/connectivity/bluetooth/bluetooth-le-host.rst [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    CONFIG_BT_SETTINGS=y
+    CONFIG_FLASH=y
+    CONFIG_FLASH_PAGE_LAYOUT=y
+    CONFIG_FLASH_MAP=y
+    CONFIG_NVS=y
+    CONFIG_SETTINGS=y
+
+Once enabled, it is the responsibility of the application to call
+settings_load() after having initialized Bluetooth (using the
+bt_enable() API).
```
Comment (theob-pro) [inline doc/connectivity/bluetooth/bluetooth-le-host.rst:204]:
```suggestion
:c:func:`bt_enable` API).
```

File: doc/connectivity/bluetooth/bluetooth-tools.rst [hunk 5]
```diff
@@ -178,6 +185,61 @@ In order to see those logs, you can use the built-in ``btmon`` tool from BlueZ:
...
+When running both Host and Controller in actual Integrated Circuits, you will
+only see normal log messages on the console by default, without any way of
+accessing the HCI traffic between the Host and the Controller.  However, there
+is a special Bluetooth logging mode that converts the console to use a binary
+protocol that interleaves both normal log messages as well as the HCI traffic.
+
+Set the following Kconfig options to enable this protocol before building your
+application:
+
+.. code-block:: console
```
Comment (theob-pro) [inline doc/connectivity/bluetooth/bluetooth-tools.rst:218]:
```suggestion
.. code-block:: cfg
```

File: doc/connectivity/bluetooth/bluetooth-le-host.rst (hunk 3 above)
Comment (jori-nordic) [reply doc/connectivity/bluetooth/bluetooth-le-host.rst:193]: fixed in a new commit

**Review by theob-pro — DISMISSED** [review]

**Comment by Thalley** [comment]
@jori-nordic does a change of > 400 lines really count as `trivial`? :D 

If it was just a move, then I'd perhaps agree, but looks like you've done several minor changes as well

**Comment by jori-nordic** [comment]
it says "like doc changes" in the label when you hover :p

**Review by alwa-nordic — DISMISSED** [review]

**Comment by Thalley** [comment]
> it says "like doc changes" in the label when you hover :p

Hah, true :D The dangers of `trivial` is that can be merged within 4 hours of a commit being opened, and it'll take me more than 4 hours to find time to review a PR of this size :D

**Comment by fabiobaltieri** [comment]
@jori-nordic this needs a rebase

File: doc/connectivity/bluetooth/api/index.rst [hunk 6]
```diff
@@ -40,3 +37,37 @@ Bluetooth APIs
...
+   hci_raw.rst
+   crypto.rst
+
+Other
+=====
+
+.. toctree::
+   :maxdepth: 1
+
+   mesh.rst
```
Comment (Thalley) [inline doc/connectivity/bluetooth/api/index.rst:71]: Hah, poor mesh, being reduce to a single point in "Other" :)

**Review by Thalley — DISMISSED** [review]

File: doc/connectivity/bluetooth/api/index.rst (hunk 6 above)
Comment (jori-nordic) [reply doc/connectivity/bluetooth/api/index.rst:71]: I really didn't know where to put it :/

**Comment by jori-nordic** [comment]
> @jori-nordic this needs a rebase

done

**Comment by jori-nordic** [comment]
@Thalley @alwa-nordic @theob-pro I'm sorry the GH UI sucks so bad. The recent pushes were only two rebases, not actual changes.

**Review by alwa-nordic — DISMISSED** [review]

File: doc/connectivity/bluetooth/api/index.rst (hunk 6 above)
Comment (Thalley) [reply doc/connectivity/bluetooth/api/index.rst:71]: I get that. I added the maintainer of the Mesh (@PavelVPV) as a reviewer as well. I think they should have a say in how Mesh should be handled in the documentation as well.

**Review by Thalley — DISMISSED** [review]

**Review by theob-pro — DISMISSED** [review]

File: doc/connectivity/bluetooth/api/index.rst (hunk 6 above)
Comment (jori-nordic) [reply doc/connectivity/bluetooth/api/index.rst:71]: yepp. keep im mind  @PavelVPV that I will have other doc PRs throughout this and next weeks. So the discussion on how to best handle and showcase mesh could go in a mesh-specific PR later :)

Comment (PavelVPV) [reply doc/connectivity/bluetooth/api/index.rst:71]: Why not put mesh under "Bluetooth Mesh" section?

Comment (jori-nordic) [reply doc/connectivity/bluetooth/api/index.rst:71]: Then I need to make a controller section also with only one link?

File: doc/connectivity/bluetooth/api/index.rst [hunk 7]
```diff
@@ -40,3 +37,44 @@ Bluetooth APIs
...
+   crypto.rst
+
+Other
+=====
+
+.. toctree::
+   :maxdepth: 1
+
+   controller.rst
+   uuid.rst
```
Comment (Thalley) [inline doc/connectivity/bluetooth/api/index.rst:80]: Is there any specific order of these?

Previously it looks like they were sorted by file name, but now that seems random. 

Not sure if it matters at all

**Review by Thalley — APPROVED** [review]

File: doc/connectivity/bluetooth/api/index.rst (hunk 6 above)
Comment (PavelVPV) [reply doc/connectivity/bluetooth/api/index.rst]: Thanks! Maybe it is the time for mesh team to revise the mesh upstream documentation.

**Review by PavelVPV — APPROVED** [review]

File: doc/connectivity/bluetooth/api/index.rst (hunk 7 above)
Comment (jori-nordic) [reply doc/connectivity/bluetooth/api/index.rst:80]: I tried to have them in stack top-down order

**Review by theob-pro — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**